pydoctor/templatewriter/summary.py

```diff
@@ -26,13 +26,26 @@ def moduleSummary(module: model.Module, page_url: str) -> Tag:
         r(class_='private')
     if not isinstance(module, model.Package):
         return r
-    contents = [m for m in module.contents.values()
-                if isinstance(m, model.Module) and m.isVisible]
+    contents = list(module.submodules())
     if not contents:
         return r
-    ul = tags.ul()
+
     def fullName(obj: model.Documentable) -> str:
         return obj.fullName()
+
+    if len(contents) > 50 and not any(any(s.submodules()) for s in contents):
+        # If there are more than 50 modules and no submodule has
+        # further submodules we use a more compact presentation.
+        pass
+        div = tags.div(class_='submodules')
```

Comment:
Maybe we could read a text like: « 50 submodules: <the list>»

Reply:
I am afraid I don't know what you mean?

Reply:
I'm suggesting to show a text saying how many modules are listed under a single list item, such that reader have a better view of the tree. Also cannot misunderstand the entry as being only one module.

Reply:
Now that the submodules are displayed in a list item the vertical bar is displayed which IMO really helps with indicating that these are modules. So I don't think adding a label is necessary.